mantidproject/mantid
# Add event handling to the reflectometry (polref) GUI

First step to implement event handling in the Reflectometry (Polref) interface.

**To test:**

- If tested on Windows, this needs to be tested in Release mode.
- Open Interfaces > Reflectometry > Polref. There should be an extra tab called `Event Handling` (with two groups).
- Load `INTER_NR_test2.tbl` (can be found in the ISIS sample dataset) and click on `Process`, you should be able to reduce the table and get the same results as before (compare results to the same process on master).
- Remove workspaces that were created in the previous step. Use the same table, but this time remove values in column `dQ/Q`. Select both groups and hit `Process`. Groups should be processed with no errors/warnings and a workspace `IvsQ_13460_13462` should have been created.
- With the same table, go to `Event Handling` and enter a numeric value, 100 for instance. Go back to `Runs` tab and click on `Process`, you should be able to reduce the runs as event data as well, this means that two workspaces called `IvsQ_13460_0_100_13462_0_100` and `IvsQ_13469_0_100_13470_0_100` should have been created.
- Open a new table (`Reflectometry` menu > `New table`). In the first and only row enter run number `38415` (make sure the selected instrument is `INTER`) and transmission run `38393`. Reduce the row and compare results with this script provided by scientists: [EventRef.zip](https://github.com/mantidproject/mantid/files/714298/EventRef.zip). `INTER` datasets:
[reflectometry_datasets.zip](https://github.com/mantidproject/mantid/files/726256/reflectometry_datasets.zip)

- Open a new table and add run `38415` in the first row. Add a new row and add run `38417`. Enter some values in column `dQ/Q`. Process the group. The first run is an event workspace but the second one is not, you should get a warning message saying that the group could not be processed as event workspace, but some workspaces should have been created in the ADS (the interface will have reduced the runs as non-event workspaces).
- Unit tests for `ReflDataProcessorPresenter` will be added in a separate issue (https://github.com/mantidproject/mantid/issues/18498).
- Documentation has been updated.


Fixes #18227.
Also fixes #18176.

Release notes: https://github.com/mantidproject/mantid/pull/18463/commits/f58a3b80390342ca308052f697233151f0b47866

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
